FABulous/timing_model/hdlnx/verilog_gate_level.py

```diff
@@ -0,0 +1,1041 @@
+                [executable, *args],
+                input=stdin_data,   
+                text=True,                   
+            )
+        else:
+            result = subprocess.run(
+                [executable, *args],
+                input=stdin_data,   
+                text=True,            
+                capture_output=True, 
+                check=False,          
+            )
+        
+        if result.returncode != 0:
+            raise RuntimeError(f"Command '{' '.join([executable, *args])}' failed with error: {result.stderr}")
+        return result
+    
+    ### Protected methods (STA connection Interface) ###
+        
+    def _generate_sdf_opensta(self) -> Path:
+        """
+        Generates an temporary SDF file from the Verilog gate-level netlist using OpenSTA.
+        The SDF file is created in a temporary location and deleted after use.
+        Returns:
+            Path: The path to the generated SDF file.
+        """
+        
+        sta_tcl_script: str = ""
+        if isinstance(self.liberty_files, Path):
+            sta_tcl_script += f"read_liberty {self.liberty_files}\n"
+        else:
+            for lib in self.liberty_files:
+                sta_tcl_script += f"read_liberty {lib}\n"
+        sta_tcl_script += f"read_verilog {self.verilog_netlist}\n"
+        sta_tcl_script += f"link_design {self.top_name}\n"
+        if self.spef_files is not None:
+            if isinstance(self.spef_files, Path):
+                sta_tcl_script += f"read_spef {self.spef_files}\n"
+            elif isinstance(self.spef_files, list):
+                for spef in self.spef_files:
+                    sta_tcl_script += f"read_spef {spef}\n"
+        sta_tcl_script += "write_sdf {}\n".format("{sdf_path}")
+        sta_tcl_script += "exit\n"
+        
+        fd, path = tempfile.mkstemp(prefix="sta_", suffix=".sdf")
+        os.close(fd) 
+        
+        if self.debug:
+            print(f"Generating SDF file at temporary path: {path}")
+        
+        self._call_external(
+            self.sta_executable,
+            stdin_data=sta_tcl_script.format(sdf_path=path),
+            debug=self.debug
+        )  
+        
+        with open(path, "r") as f:
+            content: str = f.read()
+            if len(content) == 0:
+               os.remove(path)
+               raise RuntimeError("Failed to generate SDF file using OpenSTA. No content in SDF file.")
+        
+        if path is None:
+            raise RuntimeError("Failed to generate SDF file using OpenSTA. No SDF file created.")        
+       
+        return Path(path)
+    
+    ### Public methods ###
+    
+    def get_raw_verilog_netlist_data(self) -> str:
+        """
+        Returns the raw Verilog netlist content as a string.
+        Returns:
+            str: The content of the Verilog netlist file.
+        """
+        return self.verilog_netlist_content
+
+    
+    def resolve_hier_pin(self, hier_pin_path: str) -> list[str]:
```

Comment:
Here you are trying to parse the Verilog file netlist. You can use `yosys_obj`, which should do all the parsing for you and return an object you can access easily.